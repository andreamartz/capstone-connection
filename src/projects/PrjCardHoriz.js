import React from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import Box from '@material-ui/core/Box';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Typography from '@material-ui/core/Typography';
import Tags from '../tags/Tags';

const useStyles = makeStyles((theme) => ({
  root: {
    display: 'flex',
    padding: theme.spacing(2)
  },
  details: {
    display: 'flex',
    flexDirection: 'column',
  },
  content: {
    flex: '1 0 auto',
  },
  cover: {
    width: theme.spacing(25),
  },
  tags: {
    display: 'flex',
    alignItems: 'center',
    paddingLeft: theme.spacing(1),
    paddingBottom: theme.spacing(1),
  }
}));

function PrjCardHoriz({ project }) {
  const classes = useStyles();

  return (
    <Card className={classes.root}>
      <CardMedia
        className={classes.cover}
        image={project.image}
        title={project.name}
      />
      <div className={classes.details}>
        <CardContent className={classes.content}>
          <Typography component="h5" variant="h5">
            {project.name}
          </Typography>
        </CardContent>
        <div className={classes.tags}>
          <Tags tags={project.tags}/>
        </div>
      </div>
    </Card>
  );
}

export default PrjCardHoriz;