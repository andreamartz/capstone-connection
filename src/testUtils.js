import React from "react";
import UserContext from "./auth/UserContext";

const demoUser = {
  id: 1,
  username: "testuser",
  firstName: "testfirst",
  lastName: "testlast",
  bio: "This is my bio",
  email: "test@test.net",
  photoUrl: null,
};

const UserProvider =
    ({ children, currentUser = demoUser }) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };
