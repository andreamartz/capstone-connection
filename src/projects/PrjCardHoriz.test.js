import React from 'react';
import { render } from '@testing-library/react';
import PrjCardHoriz from './PrjCardHoriz';
import { UserProvider } from '../testUtils';

const project = {
	id: 3,
	name: 'Water Mate',
	image:
		'https://res.cloudinary.com/wahmof2/image/upload/v1628103099/capstone_connections/projects_capstone_connections/water-mate-Aimee-Wildstone.png',
	tags: [{ id: 4, text: 'API' }],
};

test('it renders without crashing', function () {
	render(<PrjCardHoriz project={project} />);
});

test('it matches snapshot', function () {
	const { asFragment } = render(
		<UserProvider>
			<PrjCardHoriz project={project} />
		</UserProvider>,
	);
	expect(asFragment()).toMatchSnapshot();
});
