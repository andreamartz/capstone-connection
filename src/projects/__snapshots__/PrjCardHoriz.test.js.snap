// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-8 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardMedia-root makeStyles-cover-11"
      style="background-image: url(https://res.cloudinary.com/wahmof2/image/upload/v1628103099/capstone_connections/projects_capstone_connections/water-mate-Aimee-Wildstone.png);"
      title="Water Mate"
    />
    <div
      class="makeStyles-details-9"
    >
      <div
        class="MuiCardContent-root makeStyles-content-10"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5"
        >
          Water Mate
        </h5>
      </div>
      <div
        class="makeStyles-tags-12"
      >
        <div
          class="MuiBox-root MuiBox-root-13"
        >
          <div
            class="MuiBox-root MuiBox-root-14"
          >
            <span
              class="tag"
            >
              <p
                class="MuiTypography-root API MuiTypography-body1"
              >
                API
              </p>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
